pakyow-core/lib/pakyow/core/router.rb

```diff
@@ -196,6 +196,15 @@ class Router
 
     # @api private
     def initialize(context)
+      if router = context.previous_router_instance
+        # copy state from previous router to the new one
+        # this is so that state is shared when rerouting from one router to another one
```

Comment:
Yeah I cringed when I hit this test case. It only affects reroutes, so at least it's somewhat of a limited case. 

Calling routes in the controller means we have to do something special to make hooks/methods available to be called. I like how routes defined in a router are called in an instance of that router, because it makes things easier to understand conceptually. It also provides more helpful backtraces.

Of these two options, I settled on copying state being the best option. Any thoughts?